src/modules/Dashboard/features/RespondentData/RespondentDataReview/Feedback/FeedbackNotes/FeedbackNotes.test.tsx

```diff
@@ -114,106 +118,161 @@ const mockGetWithNotes = () => {
-        /respondents-summary-feedback-notes-note-\d+$/,
-      );
-      expect(elementsWithTestIdSubstring).toHaveLength(2);
-      expect(screen.getByText(/John Doe/)).toBeInTheDocument();
-      expect(screen.getByText(/New note/)).toBeInTheDocument();
-      expect(screen.getByText(/Jane Doe/)).toBeInTheDocument();
-      expect(screen.getByText(/Note 1/)).toBeInTheDocument();
-    });
+const testNotesRender = async () =>
+  await waitFor(() => {
+    const elementsWithTestIdSubstring = screen.queryAllByTestId(
+      /respondents-summary-feedback-notes-note-\d+$/,
+    );
+    expect(elementsWithTestIdSubstring).toHaveLength(2);
+    expect(screen.getByText(/John Doe/)).toBeInTheDocument();
+    expect(screen.getByText(/New note/)).toBeInTheDocument();
+    expect(screen.getByText(/Jane Doe/)).toBeInTheDocument();
+    expect(screen.getByText(/Note 1/)).toBeInTheDocument();
   });
 
-  test('should save new note', async () => {
-    mockGetWithNotes();
-    renderFeedbackNotes();
+const testNoteCreation = async (apiCallRoute: string) => {
+  // wait for the isLoading process of getAnswersNotes to become false
+  await waitForTheUpdate();
 
-    // wait for the isLoading process of getAnswersNotes to become false
-    await waitForTheUpdate();
+  await userEvent.type(screen.getByLabelText(/Add a New Note/i), newNoteValue);
+  await userEvent.click(screen.getByTestId('respondents-summary-feedback-notes-save'));
+  mockAxios.post.mockResolvedValueOnce(null);
 
-    await userEvent.type(screen.getByLabelText(/Add a New Note/i), newNoteValue);
-    await userEvent.click(screen.getByTestId('respondents-summary-feedback-notes-save'));
-    mockAxios.post.mockResolvedValueOnce(null);
+  await waitFor(() => {
+    expect(mockAxios.post).nthCalledWith(
+      1,
+      apiCallRoute,
+      {
+        note: newNoteValue,
+      },
+      { signal: undefined },
+    );
+  });
+};
 
-    await waitFor(() => {
-      expect(mockAxios.post).nthCalledWith(
-        1,
-        `/answers/applet/${mockedAppletId}/answers/${mockedAnswerId}/activities/${mockedActivity.id}/notes`,
-        {
-          note: newNoteValue,
-        },
-        { signal: undefined },
-      );
+const testNoteEditing = async (apiCallRoute: string) => {
+  await waitFor(() => {
+    const hiddenEditAction = screen.queryByTestId(`${noteTestId}-0-edit`);
+
+    expect(hiddenEditAction).toBeNull();
+
+    const noteHeader = screen.queryByTestId(`${noteTestId}-0-header`) as HTMLElement;
+    fireEvent.mouseEnter(noteHeader);
+    const visibleEditAction = screen.queryByTestId(`${noteTestId}-0-edit`) as HTMLElement;
+
+    expect(visibleEditAction).toBeInTheDocument();
+
+    fireEvent.click(visibleEditAction);
+  });
+
+  await waitFor(() => {
+    const noteContainer = screen.queryByTestId(`${noteTestId}-0-text`) as HTMLElement;
+    const textarea = noteContainer.querySelector('textarea') as HTMLElement;
+    fireEvent.change(textarea, {
```

Comment:
Could you please update it with `userEvent.type` here and with `userEvent.click` a couple of rows below?